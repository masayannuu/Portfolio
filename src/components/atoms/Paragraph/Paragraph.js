import React from 'react'

const Paragraph = props => {
  const getComponent = text =>
    text.split('<br>').map((paragraph, index) => <p key={index}>{paragraph}</p>)

  return <dd>{getComponent(props.text)}</dd>
}

export default Paragraph
