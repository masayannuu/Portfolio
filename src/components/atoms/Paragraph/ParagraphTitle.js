import React from 'react'
import styled from 'styled-components'

const StyledTitle = styled.p`
  margin-right: 0.8rem;
  font-feature-settings: 'palt';
  letter-spacing: 0.05em;
`

const ParagrahTitle = ({ className, children }) => (
  <StyledTitle className={className}>{children}</StyledTitle>
)

export default ParagrahTitle
