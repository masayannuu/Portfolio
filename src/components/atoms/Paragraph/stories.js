import React from 'react'
import { storiesOf } from '@storybook/react'
import ParagraphTitle from './ParagraphTitle'
import Paragraph from './Paragraph'

storiesOf('Atoms.Paragraph', module).add('ParagraphTite', () => (
  <ParagraphTitle>paragraph</ParagraphTitle>
))

storiesOf('Atoms.Paragraph', module).add('Paragraph', () => <Paragraph text="hoge<br>aaa" />)
