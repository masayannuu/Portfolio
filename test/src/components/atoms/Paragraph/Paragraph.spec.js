import React from 'react'
import Enzyme from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import Paragraph from '../../../../../src/components/atoms/Paragraph/Paragraph'

describe('Paragraph Paragraph Atoms', () => {
  it('should render Paragraph', () => {
    const wrapper = Enzyme.shallow(<Paragraph text="testです<br>改行区切り" />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
