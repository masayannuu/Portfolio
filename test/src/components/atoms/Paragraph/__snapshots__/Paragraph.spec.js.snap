// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph Paragraph Atoms should render Paragraph 1`] = `
<dd>
  <p
    key="0"
  >
    testです
  </p>
  <p
    key="1"
  >
    改行区切り
  </p>
</dd>
`;
